tests/integration/gateway/ready.test.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const config = require('../../config');
+const { api } = require('../../helpers/api');
+
+const {
+	badRequestSchema,
+	serviceUnavailableSchema,
+} = require('../../schemas/httpGenerics.schema');
+
+const {
+	readySchema,
+} = require('../../schemas/gateway/ready.schema');
+
+const baseUrl = config.SERVICE_ENDPOINT;
+const endpoint = `${baseUrl}/api/ready`;
+
+describe('Ready API', () => {
+	describe(`GET ${endpoint}`, () => {
+		it('Report readiness -> 200 OK', async () => {
+			try {
+				const response = await api.get(endpoint);
+				expect(response).toMap(readySchema);
+			} catch (_) {
+				const expectedResponseCode = 503;
+				const response = await api.get(endpoint, expectedResponseCode);
+				expect(response).toMap(serviceUnavailableSchema);
+			}
```

Comment:
Can't we only test the following case like we do for all other tests? it will also cover the test in try.
```suggestion
			const response = await api.get(endpoint, 503);
			expect(response).toMap(serviceUnavailableSchema);
```

Reply:
No, the idea here is to check for the successful response within the `try` block and in case the `/ready` endpoint is not ready it responds with a `503 SERVICE UNAVAILABLE`, which happens to be vet in the `catch` block.

If you only check for a `503 SERVICE UNAVAILABLE`, then the tests would fail if the `/ready` endpoint returns with a `200 OK` response.

Reply:
This should be fine for now, as is.